Python/Product/VSInterpreters/PackageManager/CondaUtils.cs

```diff
@@ -45,5 +53,73 @@ static class CondaUtils {
             var provider = componentModel.GetService<ICondaLocatorProvider>();
             return provider?.FindLocator()?.CondaExecutablePath;
         }
+
+        /// <summary>
+        /// Determine if an environment is created/managed by conda and thus should be activated.
+        /// </summary>
+        /// <param name="prefixPath">Path to the environment.</param>
+        /// <returns><c>true</c> if it is a conda environment.</returns>
+        internal static bool IsCondaEnvironment(string prefixPath) {
+            return Directory.Exists(Path.Combine(prefixPath, "conda-meta"));
+        }
+
+        /// <summary>
+        /// Activate the base conda environment, capture and return its environment variables
+        /// </summary>
+        /// <param name="condaPath">Path to the base conda environment's conda.exe</param>
+        /// <returns>List of environment variables.</returns>
+        internal async static Task<IEnumerable<KeyValuePair<string, string>>> GetActivatedEnvironmentVariablesAsync(string condaPath) {
+            var activateBat = Path.Combine(Path.GetDirectoryName(condaPath), "activate.bat");
+            if (File.Exists(activateBat)) {
+                using (var proc = ProcessOutput.RunHiddenAndCapture(activateBat, new[] { "&", "python.exe", "-c", PrintEnvironmentCode })) {
+                    await proc;
+                    if (proc.ExitCode == 0) {
+                        return ParseEnvironmentVariables(proc);
+                    }
+                }
+            }
+
+            return Enumerable.Empty<KeyValuePair<string, string>>();
+        }
+
+        /// <summary>
+        /// Activate the specified conda environment, capture and return its environment variables.
+        /// </summary>
+        /// <param name="condaPath">Path to the base conda environment's conda.exe</param>
+        /// <param name="prefixPath">Path to the conda environment to activate.</param>
+        /// <returns>List of environment variables.</returns>
+        internal static IEnumerable<KeyValuePair<string, string>> GetActivatedEnvironmentVariables(string condaPath, string prefixPath) {
```

Comment:
"ActivateAndGetEnvironmentVariables"? Current name implies it's already activated.

Reply:
Method name with "And" in them is usually a bad sign. I'll try to come up with something.

Reply:
I think I'll go with `CaptureActivationEnvironmentVariables`. That doesn't imply that it's already activated.